nc_files/remap-subroutines/qt_auto_probe_tool.ngc

```diff
@@ -1,89 +1,261 @@
 M6
 
-O200 if [#<_hal[qtversaprobe.enable]> EQ 0]
+; -------------------------------------
+; ------ End Physical Tool Change ----
+; -------------------------------------
+
+; -------------------------------------
+; ------ Begin Safety Checks for Tool Offset Measurement ----
+; -------------------------------------
+
+o300 if [#<_hal[qtversaprobe.enable]> EQ 0]
    (MSG, Auto Tool probe disabled )
-    G43
-O200 return [3] ; indicate no tool measurement 
-O200 endif
+    G43 ;turn back on tool offset mode before returning
+o300 return [3] ; indicate no tool measurement 
+o300 endif
+
+;check we have a usable probe velocity configured, otherwise error
+o400 if [#<_hal[qtversaprobe.probevel]> LE 0]
+o400 return [-2] ; indicate probevel <= 0 
+o400 endif
 
+; -------------------------------------
+; ------ End Safety Checks for Tool Offset Measurement ----
+; -------------------------------------
+
+; -------------------------------------
+; ------ Begin Tool Offset Measurement ----
+; -------------------------------------
+
+;rapid-move to safe height
+G53 G0 Z[#<_ini[VERSA_TOOLSETTER]Z_MAX_CLEAR>]
+;Then rapid-move to probe XY location
 G53 G0 X[#<_ini[VERSA_TOOLSETTER]X>] Y[#<_ini[VERSA_TOOLSETTER]Y>]
+;then feed down to the probe start location
 F #<_hal[qtversaprobe.searchvel]>
 G53 G1 Z[#<_ini[VERSA_TOOLSETTER]Z>]
 
+;switch to relative distance mode for probe moves
+G91
 
-O300 if [#<_hal[qtversaprobe.searchvel]> LE 0]
-O300 return [-1] ; indicate searchvel <= 0 
-O300 endif
-
-O400 if [#<_hal[qtversaprobe.probevel]> LE 0]
-O400 return [-2] ; indicate probevel <= 0 
-O400 endif
-
+;begin initial probe move
 F #<_hal[qtversaprobe.searchvel]>
-G91
 G38.2 Z- #<_ini[VERSA_TOOLSETTER]MAXPROBE>
 G0 Z #<_hal[qtversaprobe.backoffdist]>
 
-;reprobe at probe speed
+;reprobe at probe speed: G38.2 means probe toward workpiece, stop on contact
+;  when G38.n axis completes, it puts the Z value detected (in the coordinates system 
+;  in which this program started) into parameter #5063 (because we're probing Z) 
+;  
 F #<_hal[qtversaprobe.probevel]>
 G38.2 Z- [#<_hal[qtversaprobe.backoffdist]> *1.2]
 
-O500 if [#5070 EQ 0]
+o500 if [#5070 EQ 0]
 G90
-O500 return [-3] ; indicate probe contact failure to epilog
-O500 endif
+o500 return [-3] ; indicate probe contact failure to epilog
+o500 endif
 
+;switch back to absolute distance mode once probing is done
 G90
-G53 G0 Z[#<_ini[CHANGE_POSITION]Z>]
 
+;set the current offset
 #<touch_result> = #5063
 
+;
+; G10 L1 is the Set Tool Table Offset: it changes the tool table offset, 
+;   it DOES NOT change the offsets in the current coordinate system. (that would be G10 L2)
+
+G10 L1 P#<tool> Z[#<touch_result> - #<_hal[qtversaprobe.probeheight]> + #<_hal[qtversaprobe.blockheight]>]
+
+;G43 enables tool length offset: this affects all subsequent moves by applying 
+;  the offset just calculated to the coordinate system currently in
+;  effect. ie. it has subtracted Calculated Offset from the Z offset
+ 
+G43
+
```

Comment:
f your going to calculate the offset to print it - might as well use that in the G10 command .